testdata/project-v2/config/rbac/firstmate_editor_role.yaml

```diff
@@ -0,0 +1,26 @@
+# permissions to do edit firstmates.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: firstmate-editor-role
+rules:
+- apiGroups:
+  - crew.testproject.org
+  resources:
+  - firstmates
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - crew.testproject.org
+  resources:
+  - firstmates/status
+  verbs:
```

Comment:
I was comparing this with the build-in cluster role `edit` which is  you can see with `kubectl get clusterroles edit -o yaml`.  The edit role gives read and write to most of the built-in kubernetes resources.  I noted that it does not give patch or update permissions on any `*/status` resources.  Is it by design that these generated roles are different?

Reply:
If `*/status` should be included depends on who will be granted these permission.
For controllers, permissions to edit `*/status` is necessary.
For human, they should never edit `*/status`.

Reading #886 again, it seems that means human users.
I guess we can still revise it and make it clear that it is for end users.

Reply:
> I was comparing this with the build-in cluster role edit which is you can see with kubectl get clusterroles edit -o yaml.

The roles done here are to give permissions to the specific resource. I am not sure if I could understand properly the point.

> I noted that it does not give patch or update permissions on any */status resources.

I understand that the idea is to allow admin using this role to create a group of permissions. So, the status is required for the operator to update the CR's and users would also would like to check the status. Am I right? 

I am not sure if I understood what should be done in another way here. May the question could be what behaviour are you face or should not be faced when these roles are applied?